File: ruby/mandelbrot_ascii.rb
#!/usr/bin/ruby

require 'Complex'

$ascii = '.`,;:clodxkO0KXNWM'
pixel_size = 20*2
uclid_size = 4.0
asft, bsft = -0.0,-0.6
u2p = uclid_size/pixel_size
yA = (0..pixel_size).collect { |y| (y-pixel_size/2)*u2p+asft }
xA = (0..pixel_size).collect { |x| (x-pixel_size/2)*u2p+bsft }

yA.each do |y|
  xA.each do |x|
    c = oc = Complex(x,y)
    n, mn = 0, 40
    while n<mn and c.abs2<2**2 do
      c = c**2 + oc
      n += 1
    end
    print (n<mn ? $ascii[$ascii.length * Math.sin(n-Math.log(c.abs))] : ' ')*2
  end
  puts
end  

File: go/mandelbrot_ascii.go
package main
import ("fmt"; "strings"; "math/cmplx")

func iterations(z0 complex128, c int) int {
  z := z0
  for cmplx.Abs(z) < 2 && c > 0 {
	z = z * z + z0
	c--
  }
  return c
}

func main() {
  for b := -2.1; b < 2.1; b += 0.1 {
    for a := -2.1; a < 2.1; a += 0.1 {
	  fmt.Printf(strings.Repeat(string(" .,:cloO0X"[iterations(complex(a, b), 9)]), 2))
    }
    fmt.Println()
  }
}
